api/analyzers/source_analyzer.py

```
@@ -153,17 +178,71 @@ def second_pass(self, graph: Graph, files: list[Path], path: Path) -> None:
+                    _resolve_file(fp)
+                    done += 1
+                    if done % log_every == 0 or done == total:
+                        logging.info("second_pass: resolved %d/%d files", done, total)
+            else:
+                with ThreadPoolExecutor(max_workers=n_workers, thread_name_prefix="sa-resolve") as ex:
+                    futures = {ex.submit(_resolve_file, fp): fp for fp in resolvable}
+                    for fut in as_completed(futures):
+                        fp = futures[fut]
+                        try:
+                            fut.result()
+                        except Exception as exc:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)